saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequest.java

```diff
@@ -37,15 +40,19 @@
 	private final List<Saml2X509Credential> credentials;
 	private final String destination;
 	private final String assertionConsumerServiceUrl;
+	private final Saml2MessageBinding binding;
+	private final MultiValueMap<String, String> queryParameters;
 
 	private Saml2AuthenticationRequest(
 			String issuer,
 			String destination,
 			String assertionConsumerServiceUrl,
-			List<Saml2X509Credential> credentials) {
-		Assert.hasText(issuer, "issuer cannot be null");
-		Assert.hasText(destination, "destination cannot be null");
-		Assert.hasText(assertionConsumerServiceUrl, "spAssertionConsumerServiceUrl cannot be null");
+			List<Saml2X509Credential> credentials,
+			Saml2MessageBinding binding,
+			MultiValueMap<String, String> queryParameters) {
```

Comment:
Why does this need to be a `MultiValueMap`? My concern is that this leaves the door open for bugs like having more than one relay state or saml request parameter in the map. Would you be concerned if this were `Map<String, String>` or `Map<String, Object>`?

Reply:
MultiValueMap is aligned with the HTTP spec for transferring values over URLs or forms. 
I do agree, that we don't have to offer that, and it will simplify the maps.
Addressed in: d226ce2c2e4ba3af1e202186c486198d7d177f01